src/Formik.tsx

```diff
@@ -847,7 +847,7 @@ export function useFormik<Values extends FormikValues = FormikValues>({
 
   const dirty = React.useMemo(
     () => !isEqual(initialValues.current, state.values),
-    [state.values]
+    [initialValues.current, state.values]
```

Comment:
I don't think you need .current here

Reply:
It's needed. The `initialValues` reference will never change, even if some of its properties does.

There are some potential pitfalls with having refs in the dependency list. See <https://github.com/facebook/react/issues/14387> for example. This is a slightly different case since I think `resetForm` will always trigger a rerender, and this ref is not a ref to a dom node, which would have been updated after the effect hook was run.

Another solution would be to move these initial values/errors to the state instead, but I haven't looked into what that would mean for functionality and performance.